org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/commands/AssociateGrammarCommandTest.java

```diff
@@ -0,0 +1,148 @@
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for associating a grammar to a given XML command.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class AssociateGrammarCommandTest {
+
+	@Test
+	public void associateWithXSDNoNamespaceShemaLocation() throws InterruptedException, ExecutionException {
+		MockXMLLanguageServer languageServer = new MockXMLLanguageServer();
+
+		String xml = "<?xml version=\"1.0\" ?>\r\n" + //
+				"<foo/>";
+		String xmlPath = getFileURI("src/test/resources/tag.xml");
+		TextDocumentIdentifier xmlIdentifier = languageServer.didOpen(xmlPath, xml);
+		String xsdPath = getFileURI("src/test/resources/xsd/tag.xsd");
+		String bindingType = GrammarBindingType.XSD.getName();
+
+		TextDocumentEdit actual = (TextDocumentEdit) languageServer
+				.executeCommand(AssociateGrammarCommand.COMMAND_ID, xmlIdentifier, xsdPath, bindingType).get();
+		assertNotNull(actual);
+
+		assertEquals(actual, tde(xmlPath, 1, te(1, 4, 1, 4, //
+				" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+						" xsi:noNamespaceSchemaLocation=\"xsd/tag.xsd\"")));
+	}
+
+	@Test
+	public void associateWithXSDSchemaLocation() throws Exception {
+		MockXMLLanguageServer languageServer = new MockXMLLanguageServer();
+
+		String xml = "<?xml version=\"1.0\" ?>\r\n" + //
+				"<foo/>";
+		String xmlPath = getFileURI("src/test/resources/tag.xml");
+		TextDocumentIdentifier xmlIdentifier = languageServer.didOpen(xmlPath, xml);
+
+		String xsdPath = getFileURI("src/test/resources/xsd/team.xsd");
+		String bindingType = GrammarBindingType.XSD.getName();
+
+		TextDocumentEdit actual = (TextDocumentEdit) languageServer
+				.executeCommand(AssociateGrammarCommand.COMMAND_ID, xmlIdentifier, xsdPath, bindingType).get();
+		assertNotNull(actual);
+
+		assertEquals(actual, tde(xmlPath, 1, te(1, 4, 1, 4, //
+				" xmlns=\"team_namespace\"\r\n" + //
+						" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+						" xsi:schemaLocation=\"team_namespace xsd/team.xsd\"")));
+	}
+
+	@Test
+	public void associateWithDTD() throws InterruptedException, ExecutionException {
+		MockXMLLanguageServer languageServer = new MockXMLLanguageServer();
+
+		String xml = "<?xml version=\"1.0\" ?>\r\n" + //
+				"<foo/>";
+		String xmlPath = getFileURI("src/test/resources/tag.xml");
+		TextDocumentIdentifier xmlIdentifier = languageServer.didOpen(xmlPath, xml);
+		String dtdPath = getFileURI("src/test/resources/dtd/tag.dtd");
+		String bindingType = GrammarBindingType.DTD.getName();
+
+		TextDocumentEdit actual = (TextDocumentEdit) languageServer
+				.executeCommand(AssociateGrammarCommand.COMMAND_ID, xmlIdentifier, dtdPath, bindingType).get();
+		assertNotNull(actual);
+
+		assertEquals(actual, tde(xmlPath, 1, te(1, 0, 1, 0, //
+				"<!DOCTYPE foo SYSTEM \"dtd/tag.dtd\">\r\n")));
+	}
+
+	@Test
+	public void associateWithXMLModel() throws InterruptedException, ExecutionException {

```

Comment:
Add a test to bind to an XSD that has a `targetNamespace` using `xml-model`.

Reply:
Sorry I forgot to push this test, Fixed now.